src/servers/mcp-server.ts

```diff
@@ -224,6 +224,21 @@ export class MCPServer extends BaseMCPServer {
         const { question, datasourceId: sourceId } = GetAnswerSchema.parse(request.params.arguments);
 
         const answer = await this.getThoughtSpotService().getAnswerForQuestion(question, sourceId, false);
+        let tokenUrl = "";
+        
+        // Generate token and store in KV store
+        if (!answer.error && this.ctx.env?.OAUTH_KV) {
+            console.log("[DEBUG] Storing token in KV");
+            const token = crypto.randomUUID();
+            const tokenData = {
+                sessionId: answer.session_identifier,
+                generationNo: answer.generation_number,
+                instanceURL: this.ctx.props.instanceUrl,
+                accessToken: this.ctx.props.accessToken
+            };
+            await putInKV(token, tokenData, this.ctx.env);
+            tokenUrl = `${this.ctx.env?.HOST_NAME}/data/img?token=${token}`;
```

Comment:
Maybe we call `token` something like `uniqueId` or `tempId` to not make the agent believe its a token protected resource and should not be opened

Reply:
Done